// resvault-sdk/dist/resvault-sdk.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
class ResVaultSDK {
    constructor(targetOrigin = '*') {
        this.targetOrigin = targetOrigin;
        this.messageHandlers = [];
        this._messageHandler = this._handleMessage.bind(this);
    }
    sendMessage(message) {
        window.postMessage(message, this.targetOrigin);
    }
    addMessageListener(handler) {
        if (typeof handler !== 'function') {
            throw new Error('Handler must be a function');
        }
        this.messageHandlers.push(handler);
        if (this.messageHandlers.length === 1) {
            this._startListening();
        }
    }
    removeMessageListener(handler) {
        const index = this.messageHandlers.indexOf(handler);
        if (index !== -1) {
            this.messageHandlers.splice(index, 1);
        }
        if (this.messageHandlers.length === 0) {
            this._stopListening();
        }
    }
    _startListening() {
        window.addEventListener('message', this._messageHandler);
    }
    _stopListening() {
        window.removeEventListener('message', this._messageHandler);
    }
    _handleMessage(event) {
        if (event.source === window &&
            event.data &&
            event.data.type === 'FROM_CONTENT_SCRIPT') {
            for (const handler of this.messageHandlers) {
                handler(event);
            }
        }
    }
}
exports.default = ResVaultSDK;
